crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -126,7 +126,7 @@ impl From<RoomEvent> for SyncRoomEvent {
 pub struct SyncResponse {
     /// The batch token to supply in the `since` param of the next `/sync`
     /// request.
-    pub next_batch: String,
+    pub next_batch: Option<String>,
```

Comment:
This still seems wrong, can you revert it?

Reply:
Hm https://github.com/matrix-org/matrix-rust-sdk/pull/853/files/b33795910e253786971ed1202445797cbcd0313c#diff-c559762e72a5a86b43382e9ebb7f1bf5d8144f409c1789ad5bb4294eabf44d07R759 kinda depends on that. As a transaction from appservices doesnt have any next_batch token. And since everything eventually has to be this Struct the next_batch for an appservice Response would need to be Optional here. I am not sure how to otherwise solve this here :/

Reply:
I've created a new review thread at a potentially more relevant code location where I try to explain in more detail why I don't like this change. I'll resolve this thread in favor of the new one.